src/toolbar/sticky/stickytoolbarview.js

```diff
@@ -20,21 +23,79 @@ export default class StickyToolbarView extends ToolbarView {
 	/**
 	 * @inheritDoc
 	 */
-	constructor() {
-		super();
+	constructor( locale ) {
+		super( locale );
 
 		const bind = this.bind;
 
 		this.model.set( 'isSticky', false );
+		this.model.set( '_isStickyToLimiterBottom', false );
+		this.model.set( 'left', null );
+		this.model.set( 'marginLeft', null );
+		this.model.set( 'limiterElement', null );
 
 		Template.extend( this.template, {
 			attributes: {
-				// Toggle class of the toolbar when "sticky" state changes in the model.
-				class: bind.if( 'isSticky', 'ck-toolbar_sticky' )
+				class: [
+					// Toggle class of the toolbar when "sticky" state changes in the model.
+					bind.if( 'isSticky', 'ck-toolbar_sticky' ),
+					bind.if( '_isStickyToLimiterBottom', 'ck-toolbar_sticky_bottom-limit' ),
+				],
+				style: {
+					width: bind.to( 'isSticky', ( isSticky ) => {
+						// 2px compensates the border.
+						return isSticky ? toPx( this._elementPlaceholder.getBoundingClientRect().width + 2 ) : null;
+					} ),
+
+					top: bind.to( '_isStickyToLimiterBottom', ( _isStickyToLimiterBottom ) => {
+						return _isStickyToLimiterBottom ?
+							toPx( window.scrollY + this._limiterRect.bottom - this._toolbarRect.height ) : null;
+					} ),
+
+					left: bind.to( 'left' ),
```

Comment:
Scroll the page horizontally, focus editable, then scroll vertically. Now, remove `left` and `marginLeft` from the code and find out ;-)